blockchain/utxocache.go

```diff
@@ -505,12 +509,6 @@ func (s *utxoCache) spendEntry(outpoint wire.OutPoint, addIfNil *UtxoEntry) erro
 	// Mark the output as spent and modified.
 	entry.packedFlags |= tfSpent | tfModified
 
-	//TODO(stevenroose) check if it's ok to drop the pkScript
-	// Since we don't need it anymore, drop the pkScript value of the entry.
-	s.totalEntryMemory -= entry.memoryUsage()
-	entry.pkScript = nil
```

Comment:
I'll save the pprofs and post it to github but what I've noticed was `msgTx.Decode()` took up a significant size when the `utxomaxcachesize` was large enough. Profiling the master branch didn't have `msgTx.Decode()` taking up significant amount of heap.

Looking at the source code for `msgTx.Decode()` we can see that the deserialized pkscript is actually copied over to the large contiguous byte slice and the pkscript that was allocated separately is returned to the scriptPool. https://github.com/btcsuite/btcd/blob/a18c2cfbf8929b7950da34b09c1dec27646a61d6/wire/msgtx.go#L661-L676

These two bits of information gives me a high confidence that the script for a whole block is sticking around because we have a single entry that's still pointing to it. I know for a fact that go doesn't garbage collect parts of a slice as well.

> So setting to nil would actually let it be freed up?

Yeah I think that's right in that we should still be replacing with nil since once every last one of the entries stops pointing to it, that whole block's script can be gc-ed. Just really shouldn't be discounting the pkscript after we nil it out since it can still be taking up space.